packages/api/app/distribution-routes.js

```diff
@@ -19,6 +19,24 @@ function isAccessTokenExpired(accessTokenRecord) {
   return accessTokenRecord.expirationTime < Date.now();
 }
 
+/**
+ * Reads the input path and determines if this is a request for public data
+ * or not.
+ *
+ * @param {string} path - req.path paramater
+ * @returns {boolean} - whether this request goes to a public bucket
+ */
+function isPublicRequest(path) {
+  try {
+    const publicBuckets = process.env.public_buckets.split(',');
+    const requestedBucket = path.split('/').filter((d) => d).shift();
```

Comment:
What does the `.filter` do here? Is there any reason to not to do `path.split('/')[0]`?

Reply:
filters out empty values from the split array.  IF the path is `/bucket/keyvalues` . split => `["", "bucket", "keyvalues"]` . IF the path is `bucket/keyvalues` it gives `["bucket", "keyvalues"]`

So if I knew the path always had a slash, I could use `path.split('/')[1]` I suppose, do I know that's always true?

Reply:
Your approach seems safer so let's stick with that

Reply:
It's a little less clear, so I'll make it a helper function.